Tasks/Gradle/CodeAnalysis/SonarQube/endpoint.ts

```diff
@@ -1,5 +1,64 @@
-// Simple data class for a SonarQube generic endpoint
+/// <reference path="../../../../definitions/vsts-task-lib.d.ts" />
+
+import tl = require('vsts-task-lib/task');
+
 export class SonarQubeEndpoint {
     constructor(public Url: string, public Username: string, public Password: string) {
     }
+
+    public static createSonarQubeEndpoint():SonarQubeEndpoint {
+        var errorMessage = "Could not decode the generic endpoint. Please ensure you are running the latest agent (min version 0.3.2)";
```

Comment:
Resolved.

Removed all checks except for _if (tl.getEndpointUrl == null) ..._, which throws an empty error and makes a debug log.